node/src/neighborhood/mod.rs

```
@@ -259,16 +259,29 @@ impl Handler<ConnectionProgressMessage> for Neighborhood {
     type Result = ();
 
     fn handle(&mut self, msg: ConnectionProgressMessage, ctx: &mut Self::Context) -> Self::Result {
-        self.overall_connection_status.update_connection_stage(
-            msg.peer_addr,
-            msg.event.clone(),
-            self.node_to_ui_recipient_opt
-                .as_ref()
-                .expect("UI Gateway is unbound"),
-        );
-
-        if msg.event == ConnectionProgressEvent::TcpConnectionSuccessful {
-            self.send_ask_about_debut_gossip_message(ctx, msg.peer_addr);
+        if let Ok(connection_progress) = self
+            .overall_connection_status
+            .get_connection_progress_by_ip(msg.peer_addr)
+        {
+            OverallConnectionStatus::update_connection_stage(
+                connection_progress,
+                msg.event.clone(),
+            );
+            match msg.event {
+                ConnectionProgressEvent::TcpConnectionSuccessful => {
+                    self.send_ask_about_debut_gossip_message(ctx, msg.peer_addr);
+                }
+                ConnectionProgressEvent::IntroductionGossipReceived(_)
+                | ConnectionProgressEvent::StandardGossipReceived => {
+                    self.overall_connection_status
+                        .update_ocs_stage_and_send_message_to_ui(
+                            self.node_to_ui_recipient_opt
+                                .as_ref()
+                                .expect("UI Gateway is unbound"),
+                        );
+                }
+                _ => (),
```

You don't want to swallow the error message. Maybe log it?